packages/cli/src/commands/test.js

```diff
@@ -1,20 +1,46 @@
 import execa from 'execa'
-import Listr from 'listr'
-import VerboseRenderer from 'listr-verbose-renderer'
 import terminalLink from 'terminal-link'
 
 import { getPaths } from 'src/lib'
 import c from 'src/lib/colors'
 
+const jest = require('jest')
+
+// TODO: Get from redwood.toml
+const sides = ['web', 'api']
+
+// https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/test.js#L39
+function isInGitRepository() {
+  try {
+    execa.commandSync('git rev-parse --is-inside-work-tree')
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+function isInMercurialRepository() {
+  try {
+    execa.commandSync('hg --cwd . root')
+    return true
+  } catch (e) {
+    return false
+  }
+}
```

Comment:
This is weird, the type of repository you're using matters to jest --watch?

Reply:
I think internally in Jest it matters, but for us, no. These functions are just to check if there is _any_ version control going on, because otherwise running `watch` doesn't make sense.

(I took this from the `react-scripts` test command).

Reply:
To be clear: this is for convenience, it's definitely not required for anything to work!

Reply:
Oh that's interesting, so jest will somehow figure out which tests are not committed?

Reply:
Oh wow I missed all these follow-up comments 5 days ago, whoops!

Exactly! Jest will check which files are unstaged, check which tests are related to those files (and to tests which are related to the files that are related to/imported by those files) and run 'em. This is why running `--watch` doesn't make sense when there's no staged files, or no version control at all.

Reply:
That's pretty amazing! Nice!